src/trading/getQuote.ts

```diff
@@ -104,20 +109,63 @@ export async function getQuote(
 
   log('Getting quote...')
 
-  const quoteResponse = await orderBookApi.getQuote(quoteRequest)
+  const quote = await orderBookApi.getQuote(quoteRequest)
+
+  return {
+    quote,
+    appDataInfo,
+    orderBookApi,
+    tradeParameters,
+    slippageBpsToUse,
+  }
+}
+
+export async function getQuote(
+  _tradeParameters: TradeParameters,
+  trader: QuoterParameters,
+  advancedSettings?: SwapAdvancedSettings,
+  _orderBookApi?: OrderBookApi
+): Promise<{ result: QuoteResults; orderBookApi: OrderBookApi }> {
+  const { quote, orderBookApi, tradeParameters, slippageBpsToUse, appDataInfo } = await getQuoteRaw(
+    _tradeParameters,
+    trader,
+    advancedSettings,
+    _orderBookApi
+  )
+  const { partnerFee, slippageBps, sellTokenDecimals, buyTokenDecimals } = tradeParameters
+  const { chainId, account: from } = trader
+
+  // If AUTO slippage is used, we need to calculate the suggested slippage based on the quote)
+  if (slippageBps === undefined) {
+    const suggestedSlippageBps = suggestSlippageBps({ quote, tradeParameters, trader, advancedSettings })
+
+    // If suggested slippage is greater than default, we use the suggested slippage
+    if (suggestedSlippageBps > DEFAULT_SLIPPAGE_BPS) {
+      // Recursive call, this time using the suggested slippage
+      log(
+        `Suggested slippage is greater than ${DEFAULT_SLIPPAGE_BPS} BPS (default), calling getQuote again with suggested slippage=${suggestedSlippageBps}`
+      )
+      const newTradeParameters = { ..._tradeParameters, slippageBps: suggestedSlippageBps }
+      return getQuote(newTradeParameters, trader, advancedSettings, orderBookApi)
```

Comment:
Should `getQuoteRaw` to be used here instead of `getQuote`?

Reply:
No, here we do a recursion to the main function. 

We didn't have a `slippageBps` (we are in AUTO slippage), but now we have a suggestion. 
So we do a last recursive call, to return the quote for the new slippage. 

Probably this is a bit optimizable though, cause it will do a second API call. I will try to optimize.